go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/filecoin-project/go-indexer-core
 go 1.15
 
 require (
+	github.com/adlrocha/mapmutex v0.0.0-20200716162114-c133e97096b7
```

Comment:
The map mutex is strange in that it attempts to get the lock, but that could actually fail after enough failed attempts.  Also, the fact that it has a Sleep is a concern.  If we need a lock that can timeout if not acquired quickly enough then maybe it should wait on a context instead.

Here is another implementation for a keyed mutex to consider:
https://github.com/im7mortal/kmutex